README.md

```diff
@@ -0,0 +1,661 @@
+...     deltas = [abs(42 - val) for val in vals]
+...     return vals[deltas.index(min(deltas))]
+
+>>> specs = {
+...     "value_1": Tag("METER_A"),
+...     "value_2": Tag("METER_B"),
+...     "value_3": Tag("METER_C"),
+...     "answer_to_everything": Tag.call(closest_to_42, Tag("METER_A"), Tag("METER_B"), Tag("METER_C")),
+... }
+>>> TagResolver(retrieval_function).resolve(specs)
+{
+    "value_1": 11,
+    "value_2": 44,
+    "value_3": 57,
+    "answer_to_everything": 44,
+}
+```
+
+#### Referencing Functions by Name
+
+Both `calc` and `call` accept a string instead of a callable for their first argument. In this case, the string
+must match a key in a dict passed to `TagResolver`. This dict contains the actual callables which are then used as
+described above.
+
+This feature can be used to address issues with importing Python modules.
+
+``` python
+>>> my_extension = {
+...     "galons_per_minute": lambda val: return val * 4.40287,
+... }
+
+>>> specs = {
+...     "flow_in_sm3_per_hour": Tag("FLOW_METER.123"),
+...     "flow_in_galons_per_minute": Tag("FLOW_METER.123").calc("galons_per_minute"),
+... }
+
+>>> TagResolver(retrieval_function, my_extension).resolve(specs)
+{
+    "flow_in_sm3_per_hour": 12.3456,
+    "flow_in_galons_per_minute": 54.35604149,
+}
+```
+
+``` python
+>>> def closest_to_42(*vals):
+...     """Return whichever value in `vals` is closest to 42"""
+...     deltas = [abs(42 - val) for val in vals]
+...     return vals[deltas.index(min(deltas))]
+
+>>> specs = {
+...     "value_1": Tag("METER_A"),
+...     "value_2": Tag("METER_B"),
+...     "value_3": Tag("METER_C"),
+...     "answer_to_everything": Tag.call("nearest_42", Tag("METER_A"), Tag("METER_B"), Tag("METER_C")),
+... }
+
+>>> TagResolver(retrieval_function, {"nearest_42": closest_to_42}).resolve(specs)
+{
+    "value_1": 11,
+    "value_2": 44,
+    "value_3": 57,
+    "answer_to_everything": 44,
+}
+```
+
+### Caching and Combined API Calls
+
+Any call to `TagResolver.resolve` will result in the minimum number of calls to the API to retrieves all needed values.
+
+Each instance of `TagResolver` keeps internal cache and only queries the API for tags that are needed.
+
+In the next example with multiple calls to `resolve()`, the CDF time series API endpoint is hit only once.
+
+``` python
+>>> resolver = TagResolver(retrieval_function)
+
+>>> resolver.resolve({
+...     "value_1": Tag("METER_A"),
+...     "value_2": Tag("METER_B"),
+...     "value_3": Tag("METER_C"),
+...     "val_1_and_2": Tag("METER_A") + Tag("METER_B"),
+...     "val_2_and_3": Tag("METER_B") + Tag("METER_C"),
+...     "val_1_and_3": Tag("METER_A") + Tag("METER_C"),
+... })
+{"value_1": 12, "value_2": 23, "value_3": 34, "val_1_and_2": 35, "val_2_and_3": ...}
+
+>>> resolver.resolve({
+...     "value_1": Tag("METER_A"),
+...     "value_1_percent": 100 * Tag("METER_A") / (Tag("METER_A") + Tag("METER_B") + Tag("METER_C")),
+... })
+{"value_1": 12, "value_1_percent": 17,3913043478}
```

Comment:
```suggestion
{"value_1": 12, "value_1_percent": 17.3913043478}
```

Reply:
I was copy-pasting these values from terminal, how on Earth did I wind up with all those commas in place of points?! o_O
Thanks for finding them!